File: dates.cpp
#include <iostream>

int main(void) {
  int last_day[12]{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
  int n;
  std::cin >> n;
  for (int i = 0; i < n; i++) {
    int day, month, year;
    std::cin >> day >> month >> year;
    if (day <= last_day[month - 1]) {
      std::cout << "YES\n";
    } else {
      if (month == 2 &&
          ((year % 4 == 0 && year % 100 != 0) || year % 400 == 0) && day == 29)
        std::cout << "YES\n";
      else
        std::cout << "NO\n";
    }
    std::fflush(stdout);
  }
  return 0;
}

File: dates.py
last_day = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
n = int(input())
for _ in range(n):
    day, month, year = map(int, input().split())
    if day <= last_day[month - 1]:
        print("YES")
    else:
        if month == 2 and ((year % 4 == 0 and year % 100 != 0) or year % 400 == 0) and day == 29:
            print("YES")
        else:
            print("NO")
